hooks/useNewMqlQuery.ts

```diff
@@ -0,0 +1,428 @@
+import { useEffect, useReducer, useContext } from "react";
+import { CombinedError } from "urql";
+
+import MqlContext from "../context/MqlContext/MqlContext";
+import CreateMqlQueryWithTimeGranularity from "../mutations/mql/CreateMqlQueryWithTimeGranularity";
+import {
+  CreateMqlQueryWithTimeGranularityMutation,
+  CreateMqlQueryWithTimeGranularityMutationVariables,
+  ConstraintInput,
+  TimeGranularity,
+  Granularity
+} from "../mutations/mql/MqlMutationTypes";
+import FetchMqlQueryTimeSeries from "../queries/mql/FetchMqlQueryTimeSeries";
+import {
+  FetchMqlTimeSeriesQuery,
+  FetchMqlTimeSeriesQueryVariables,
+  MqlQueryStatus,
+} from "../queries/mql/MqlQueryTypes";
+
+const granularityToTimeGranularityMap = {
+  [Granularity.Daily]: TimeGranularity.Day,
+  [Granularity.Weekly]: TimeGranularity.Week,
+  [Granularity.Monthly]: TimeGranularity.Month,
+  // [Granularity.Daily]: TimeGranularity.Quarter,
```

Comment:
Does it make sense to change the commented key name to something other than `[Granularity.Daily]` so we don't inadvertently end up with a duplicate key down the road?